### PR TITLE
feat: add create repo with github template task

### PR DESCRIPTION
**Pull request by anupbaranwal**
# Description

Closes : 

# Checklist:

- [ ] My code follows the contribution guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] My commits follow [conventional commit message guidelines](https://www.conventionalcommits.org/en/v1.0.0/)

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -109,6 +111,29 @@ gulp.task('cleanup', () => {
   ]);
 });
 
+gulp.task('create-repo', () => {
+  axios({
+    method: 'post',
+    url:'/basic-template-repository/generate',
+    baseURL: 'https://api.github.com/repos/anupbaranwal',
+    headers: {
+      'Authorization': 'Bearer 5ca06fea7ad038374ab22aa72088fb4414bc12ad',
```
Comment (paul58914080) [inline gulpfile.js:120]: This is a security leak

File: gulpfile.js [hunk 2]
```diff
@@ -109,6 +111,29 @@ gulp.task('cleanup', () => {
...
+      'Accept': 'application/vnd.github.baptiste-preview+json'
+    },
+    data: {
+      owner: 'anupbaranwal',
+      name: domainNameLowerCase,
+      description: 'test-repo',
```
Comment (paul58914080) [inline gulpfile.js:126]: Need to have a better naming.

File: gulpfile.js [hunk 3]
```diff
@@ -109,6 +111,29 @@ gulp.task('cleanup', () => {
...
+      private: false
+    }
+  }).then(res => {
+    console.log('success', res);
+  }).catch(error => {
+    console.log('Anup error' + error);
```
Comment (paul58914080) [inline gulpfile.js:132]: This seems an error by itself .. lol

File: gulpfile.js [hunk 4]
```diff
@@ -109,6 +111,29 @@ gulp.task('cleanup', () => {
   ]);
 });
 
+gulp.task('create-repo', () => {
+  axios({
+    method: 'post',
+    url:'/basic-template-repository/generate',
```
Comment (paul58914080) [inline gulpfile.js:117]: https://developer.github.com/v3/repos/#create-a-repository-using-a-template

File: gulpfile.js [hunk 5]
```diff
@@ -8,88 +8,90 @@ const assertOk = require('assert-ok');
...
+  `!${__dirname}/templates/${templateName}/domain-api/.idea/**`,
+  `!${__dirname}/templates/${templateName}/rest-adapter/.idea/**`,
+  `!${__dirname}/templates/${templateName}/jpa-adapter/.idea/**`,
+  `!${__dirname}/templates/${templateName}/bootstrap/.idea/**`,
+];
+
+const pathSrc = [
+  `${__dirname}/templates/${templateName}/**`, // template path
+  ...filesTobeIgnored, // do not scan intellij files,
+];
```
Comment (paul58914080) [inline gulpfile.js:88]: There is no need to move this all of it to the top as global. Maybe some. How about we have an init() and the initialise things like domainNameUpper case etc create an object which can be used by the tasks ?

**Review by paul58914080 — CHANGES_REQUESTED** [review]

File: gulpfile.js (hunk 4 above)
Comment (anupbaranwal) [reply gulpfile.js:117]: It's the same api I am using.

Comment (paul58914080) [reply gulpfile.js:117]: are you sure ??

`POST /repos/:template_owner/:template_repo/generate`

You did something like this

`https://api.github.com/repos/anupbaranwal/basic-template-repository/generate`

The template owner is `devs-from-matrix` no?

Comment (paul58914080) [reply gulpfile.js:117]: I get it, you have forked that project too. 

You can ignore my comment

**Pull request closed without merging**